cpp/src/arrow/ipc/writer.cc

```diff
@@ -222,8 +222,9 @@ class RecordBatchSerializer {
       RETURN_NOT_OK(
           result->Resize(actual_length + sizeof(int64_t), /* shrink_to_fit= */ true));
     }
-    *reinterpret_cast<int64_t*>(result->mutable_data()) =
-        bit_util::ToLittleEndian(prefixed_length);
+    int64_t prefixed_length_little_endian = bit_util::ToLittleEndian(prefixed_length);
```

Comment:
you can use utils at https://github.com/apache/arrow/blob/main/cpp/src/arrow/util/ubsan.h#L81

Reply:
done